README.md

```diff
@@ -25,38 +61,172 @@ By using this handler, you can now use CDK templates with all your favorite Scep
 
 The template "type" for this handler is `cdk`. 
 
+### Deployment Types
+The CDK Handler supports two different deployment types, which function somewhat differently. These
+are "bootstrapped" and "bootstrapless", both passed to the `deployment_type` template handler argument.
+
+#### The "bootstrapped" deployment_type
+The "bootstrapped" deployment type is a more typical CDK-like way to deploy infrastructure and 
+conforms more to the standard ways _CDK_ operates. If your organization has a lot of CDK infrastructure,
+using this deployment_type likely will allow Sceptre to interoperate with existing patterns and 
+policies. 
+
+The `"bootstrapped"` deployment_type causes Sceptre assets (namely S3-destined files and 
+ECR-destined images) to be deployed using the usual CDK-bootstrapped machinery. Specifically, by 
+referencing a "qualifier", CDK looks for a corresponding stack in the AWS account that contains the
+specific S3 bucket and ECR repo, as well as IAM roles to be assumed in order build and push those
+assets up to the cloud.
+
+In order to use the "bootstrapped" deployment type to push assets to the cloud, a CDK bootstrap stack 
+with matching qualifier must already be deployed. It may be deployed via CDK (outside of Sceptre) or
+you can use CDK to generate the bootstrap template for Sceptre to deploy using  
+`cdk bootstrap --show-template > cdk-bootstrap.yaml`. It is recommended, if not reusing an existing
+bootstrap stack, to deploy the bootstrap stack using Sceptre, as it will allow you to add outputs to
+the template and reference those when setting up your CDK-based StackConfigs.
+
+With that said, a bootstrap stack is not actually necessary if your stack includes no S3 or ECR 
+assets to push.
+
+**Important:** See section below on IAM implications and behavior of using this handler and how it
+corresponds to the roles in the bootstrap stack.
+
+#### The "bootstrapless" deployment_type
+While the "bootstrapped" deployment_type is more similar to CDK's way of operating, it's less typical
+for Sceptre. Sceptre shines by allowing you to define infrastructure in different stacks and then
+"wire them together" using powerful (and very handy) hooks and resolvers. Thus, a more "Sceptre-like"
+way would be to avoid using the CDK bootstrap stack and just providing references to the required
+asset-related infrastructure with resolvers like `!stack_output`.
+
+The "bootstrapless" deployment_type uses the [cdk-bootstrapless-synthesizer](https://github.com/aws-samples/cdk-bootstrapless-synthesizer)
```

Comment:
is there any setup required by the user to allow the "bootrapless" deployment to use the cdk-boostrapless-synthesizer?

Reply:
Not if there are no assets. If there _are_ file or image assets, you will need to create the S3 bucket and/or ECR repo (along with possibly the roles you'd want to use to push them) and then specify them on the bootstrapless config.

Reply:
> i'm assuming ... If there are file or image assets, you will need to create the S3 bucket and/or ECR repo (along with possibly the roles you'd want to use to push them) means deploy cdk bootstrap --show-template > cdk-bootstrap.yaml, correct?

No. It means supply your own, however you want to supply them. Just like we do with everything else in Sceptre. 

Personally, I would want to use CDK for programmatic generation of templates with Python. I would want to create my own ECR repo and bucket to use for assets and then use my deployment `iam_role` to push assets. I wouldn't want to mess around with a bootstrap stack at all. I'd want to output those values from other stacks and reference them using `!stack_output`. 

The bootstrap stack always creates a singular bucket, a singular ECR repo, and 3 or 4 IAM roles. These roles are used for:
* The cloudformation service role (I avoid these, personally, since they're permanent and cannot be removed)
* The role used to create and execute cloudformation change sets (assumed by CDK pipelines or individual users)
* the role used to push file assets to S3
* the role used to push images to ECR

The way I've made this handler, it's up to the user to determine if they want to use the CDK bootstrap stack (granted, Sceptre won't use the CF service role or execution role unless you explicitly set them for the `role_arn` and `iam_role` configurations, respectively) or if the user wants to supply just the things they want via the "normal" Sceptre way of passing values (using !stack_output to wire stacks together).

Reply:
yeah, i know what you mean however that's not the easiest nor most straight forward path to getting all this to work.  I'm wondering whether it would make sense to provide a more prescriptive way to setup for the bootrapless mode?  to me it seems like the simplest way is to deploy "cdk bootstrap --show-template > cdk-bootstrap.yaml".  if you prefer not to provide that then maybe you can be more explicit about what's expected before running bootrapless?  the above description is good or maybe just reference existing AWS docs?

Reply:
I'll put some thought into how to reword these docs. I see your point.

Reply:
I've clarified this point in the wiki.